x/permission/keeper/keeper.go

```diff
@@ -384,6 +384,10 @@ func (k Keeper) ForceDeleteAccountPolicyForResource(ctx sdk.Context, maxDelete,
 	defer iterator.Close()
 
 	for ; iterator.Valid(); iterator.Next() {
+		// if exceeding the limit, pause the GC and mark the current resource's deletion is not complete yet
+		if deletedTotal >= maxDelete {
+			return deletedTotal, false
+		}
```

Comment:
what is the motivation to move up these code?

Reply:
How about make a variable of `ctx.IsUpgraded(upgradetypes.Nagqu)` out of the iteration?

Reply:
as well as other places